src/components/FullDetailView/FullDetailPage.js

```diff
@@ -57,71 +67,118 @@ export class FullDetailPage extends Component {
 
   componentWillReceiveProps(nextProps) {
     const { path, component } = nextProps
-    if (path && path !== this.props.path) this.handleNewSpec(component)
+
+    if (path && path !== this.props.path) {
+      if (component.changes) {
+        this.setState({ changes: component.changes }, () => this.handleNewSpec(component))
+      } else this.handleNewSpec(component)
+    }
   }
 
   // Get the data for the current definition
   handleNewSpec(component) {
     const { token, uiInspectGetDefinition, uiInspectGetCuration, uiInspectGetHarvested } = this.props
-    if (component) {
-      uiInspectGetDefinition(token, component)
-      uiInspectGetCuration(token, component)
-      uiInspectGetHarvested(token, component)
-    }
+    if (!component) return false
```

Comment:
does a value need to be returned here? A bit misleading unless it is meant to mean something.